### PR TITLE
Ft cycle delete

### PR DESCRIPTION
**Pull request by capitale250**
## What does this Pr do?

check before delete cycle

## Description of Task to be completed?

> to check if cycle has traines in it before  being deleted( cycle delete)

##How should this be manually tested?

>run `mutation DeleteApplicationCycle`

Screen Shot 2022-11-03 at 17 19 45
Screen Shot 2022-11-03 at 17 17 22
Screen Shot 2022-11-03 at 17 13 53

### CODE REVIEW

File: src/resolvers/applicationCycleResolver.ts [hunk 1]
```diff
@@ -21,10 +22,19 @@ const applicationCycleResolver:any={
...
               },
               async deleteApplicationCycle(_parent:any, _args:any){
-                const deleteapplicationCycle= await applicationCycle.findById(_args.id);
-                if(!deleteapplicationCycle) throw new Error("This applicationCycle doesn't exist")
-                const deletedapplicationCycle = await applicationCycle.findByIdAndRemove(_args.id);
-                return deletedapplicationCycle;
+                      const deleteapplicationCycle= await applicationCycle.findById(_args.id);
+                      if(deleteapplicationCycle !=null){
+                          const name=deleteapplicationCycle.name
+                          const users = await  traineEAttributes.findOne({cohort:deleteapplicationCycle.name})
```
Comment (mugiisha) [inline src/resolvers/applicationCycleResolver.ts:28]: currently the cycle( previously named cohort) is in Trainee model not in traineeAttributes model you can check updates on develop and update this one

Comment (mugiisha) [inline src/resolvers/applicationCycleResolver.ts]: currently cycle(previously named cohort) is in Trainee model and removed from traineeAttributes model, check updates on develop and update this one

**Review by mugiisha — CHANGES_REQUESTED** [review]

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
const applicationCycleToDelete= await applicationCycle.findById(_args.id);
tryb to rename variables to be meaningful

**Comment by capitale250** [comment]
> applicationCycleToDelete

gone

File: src/resolvers/applicationCycleResolver.ts [hunk 2]
```diff
@@ -21,10 +23,18 @@ const applicationCycleResolver:any={
...
               async deleteApplicationCycle(_parent:any, _args:any){
-                const deleteapplicationCycle= await applicationCycle.findById(_args.id);
-                if(!deleteapplicationCycle) throw new Error("This applicationCycle doesn't exist")
-                const deletedapplicationCycle = await applicationCycle.findByIdAndRemove(_args.id);
-                return deletedapplicationCycle;
+                      const applicationCycleToDelete= await applicationCycle.findById(_args.id);
+                      if(applicationCycleToDelete !=null){
+                          const users = await  TraineeApplicant.findOne({cycle_id:_args.id})
+                          if (users) {
+                                throw new Error("`cycle has applicant in it`")
```
Comment (istafrica-dennw) [inline src/resolvers/applicationCycleResolver.ts]: 1. `cycle has applicant in it' this string should change to
cycle has some applicants.
2. const users ===? const user (you are finding one)

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

File: src/resolvers/applicationCycleResolver.ts (hunk 2 above)
Comment (capitale250) [reply src/resolvers/applicationCycleResolver.ts]: done

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**